packages/core/js-client/src/particle/Particle.ts

```diff
@@ -47,12 +52,27 @@ export class Particle implements IParticle {
             toUint8Array(json.data),
             json.ttl,
             json.init_peer_id,
+            toUint8Array(json.signature)
         );
 
         return res;
     }
 }
 
+/**
+ * Builds particle message for signing
+ */
+export const buildParticleMessage = ({ id, timestamp, ttl, script }: Omit<IParticle, 'initPeerId' | 'signature' | 'data'>): Uint8Array => {
+    const en = new TextEncoder();
```

Comment:
you can move this part outside, I believe there is no need to recreate TextEncoder all the time

Reply:
Yes, i always think about it. In my opinion TextEncoder is a lightweight class, there is no perf implications for recreating it on a function call as far as i know. Probably keeping it like this is fine. Moving it outside means polluting module scope

Reply:
Actually, specification says that this constructor does just nothing
<img width="1347" alt="image" src="https://github.com/fluencelabs/js-client/assets/59872966/27721188-846a-43e7-af25-0e95360b09ad">

Reply:
It's still a new object each time. I don't think it wil measurably matter, but why not do it in a genrally better way since it cost us nothing to do a bit better - we just need to move this line a bit higher - that's it. Code won't suffer from that change at all as well

Reply:
ok, refactored this a s you suggested